test/library/packages/csv/CSV.chpl

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Support for reading and writing Comma Separated Value files.
+
+   This module provides the :record:`CSVIO` type for reading and writing
+   CSV files.
+ */
+module CSV {
+  /* The `CSVIO` record can be initialized with a reader or writer channel
+     for reading or writing CSV files.
+   */
+  record CSVIO {
+    var ch: channel;
+    var sep: string;
+    var hasHeader: bool;
+
+    class ReaderWriterMismatchError: Error {}
+
+    /* Initialize a CSVIO record.
+       :arg ch: The channel to read from or write to.
+       :arg sep: (optional) The delimiter to separate fields
+       :arg hasHeader: (optional) If true, treat the first line of data
+                       as a header and skip it
+     */
+    proc init(ch: channel, sep: string=",", hasHeader: bool=false) {
+      this.ch = ch;
+      this.sep = sep;
+      this.hasHeader = hasHeader;
+    }
+    /* Read a CSV file with lines matching the types of the fields in a record
+     */
+    iter readCSV(type t) throws where isRecordType(t) {
```

Comment:
I like `readCSV` a lot better than `readRecordCSV`, but now I'm wondering why not just `CSVIO.read()`? (Sorry for the shape-shifting review)